bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/BLockManager.java

```diff
@@ -0,0 +1,53 @@
+/*
+ *   Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *   WSO2 Inc. licenses this file to you under the Apache License,
+ *   Version 2.0 (the "License"); you may not use this file except
+ *   in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing,
+ *   software distributed under the License is distributed on an
+ *   "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *   KIND, either express or implied.  See the License for the
+ *   specific language governing permissions and limitations
+ *   under the License.
+ */
+
+package org.ballerinalang.jvm;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ *
+ * @since 1.2.0
+ */
+public class BLockManager {
+
+    /**
+     * The map of locks inferred.
+     */
+    private static Map<String, BLock> globalLockMap;
+
+    static {
+        globalLockMap = new HashMap<>();
```

Comment:
I think we can use a concurrent hashmap here, and get rid of manual synchronization.
https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/ConcurrentHashMap.html

Reply:
+1